packages/contracts/test/contracts/OVM/chain/OVM_CanonicalTransactionChain.spec.ts

```diff
@@ -556,6 +556,67 @@ describe('OVM_CanonicalTransactionChain', () => {
       ).to.equal(true)
     })
 
+    it('should reject a valid queue transaction if QueueOrigin is modified to SEQUENCER_QUEUE', async () => {
+      const entrypoint = NON_ZERO_ADDRESS
+      const gasLimit = 500_000
+      const data = '0x' + '12'.repeat(1234)
+
+      const timestamp = (await getEthTime(ethers.provider)) + 100
+      await setEthTime(ethers.provider, timestamp)
+      await OVM_CanonicalTransactionChain.enqueue(entrypoint, gasLimit, data)
+
+      const blockNumber = await ethers.provider.getBlockNumber()
+
+      await appendSequencerBatch(
+        OVM_CanonicalTransactionChain.connect(sequencer),
+        {
+          shouldStartAtElement: 0,
+          totalElementsToAppend: 1,
+          contexts: [
+            {
+              numSequencedTransactions: 0,
+              numSubsequentQueueTransactions: 1,
+              timestamp,
+              blockNumber,
+            },
+          ],
+          transactions: [],
+        }
+      )
+
+      await expect(
+        OVM_CanonicalTransactionChain.verifyTransaction(
+          {
+            timestamp,
+            blockNumber,
+            l1QueueOrigin: 0, // QueueOrigin.SEQUENCER_QUEUE = 0
```

Comment:
This isn't defined as an enum someplace?

Reply:
It was in another file. I just added a small [refactor commit](https://github.com/ethereum-optimism/optimism/pull/1155/commits/628c3fa8d3175055f55dc26eff58d3f298d61844) to move it into the helpers